source/extensions/common/wasm/context.cc

```diff
@@ -444,12 +442,28 @@ enum class PropertyToken { PROPERTY_TOKENS(_DECLARE) };
 static absl::flat_hash_map<std::string, PropertyToken> property_tokens = {PROPERTY_TOKENS(_PAIR)};
 #undef _PAIR
 
+absl::optional<google::api::expr::runtime::CelValue>
+Context::FindValue(absl::string_view name, Protobuf::Arena* arena) const {
+  return findValue(name, arena, false);
+}
+
 absl::optional<google::api::expr::runtime::CelValue>
 Context::findValue(absl::string_view name, Protobuf::Arena* arena, bool last) const {
   using google::api::expr::runtime::CelProtoWrapper;
   using google::api::expr::runtime::CelValue;
 
   const StreamInfo::StreamInfo* info = getConstRequestStreamInfo();
+  activation_info_ = info;
```

Comment:
Is there a way to improve this, so that the `mutable` business in the StreamActivation is not needed. It looks from this code that you need to set the members of the StreamActivation to make the FindValue work. Would it be possible to add another method that accepts these header maps and stream inf as parameters, so you do not need to have to flip StreamActivation class members? It looks like the fact that FindValue is virtual is not really useful here since you know what method is being called here.

/wait-any

Reply:
Done.